components/FileOverview.js

```diff
@@ -1,87 +1,58 @@
 import FileIcon from './FileIcon.js';
+import PackageIcon from './PackageIcon.js';
 import { SearchInput } from './SearchInput.js';
-
 import { styles as fileStyles } from './PackageOverview.js';
 
-const FileList = ({ title, files, packageName, filter }) => html`
+const FileList = ({ title, files, packageName }) => html`
   <div key=${title}>
     <h2>${title}</h2>
-    <small>${files.length} Files</small>
+    <small>${Object.entries(files).length} Files</small>
   </div>
-  <ul className=${fileStyles.directory} key=${files.join('-')}>
-    ${files
-      .filter(x => x.url.match(filter))
-      .map(
-        x => html`
-          <li key=${x.url} data-test="Item">
-            ${FileIcon}
-            <${Link} href=${x.url.replace('https://unpkg.com/', '/?')}>
-              ${x.url
-                .replace(`https://unpkg.com/`, '')
-                .replace(packageName, '')}
-            <//>
-            <small>${formatBytes(x.size)}</small>
-          </li>
-        `
-      )}
+  <ul className=${fileStyles.directory}>
+    ${Object.entries(files).map(
+      ([key, url]) => html`
+        <li key=${key} data-test="Item">
+          ${url.includes(packageName) ? FileIcon : PackageIcon}
+          <${Link} href=${url.replace('https://unpkg.com/', '/?')}>
+            ${url.replace(`https://unpkg.com/`, '').replace(packageName, '')}
+          <//>
+        </li>
+      `
+    )}
   </ul>
 `;
 
-export const FileOverview = ({ file, dispatch }) => {
-  const [cache, setCache] = react.useState({});
+export const FileOverview = () => {
+  const [{ request, cache }] = useStateValue();
   const [searchTerm, setSearchTerm] = react.useState('');
-
-  react.useEffect(() => {
-    if (file.url) {
-      console.log(`Analysing ${file.url}`);
-      recursiveDependencyFetch(file.url).then(x => {
-        setCache(x);
-        dispatch({ type: 'setDependencies', payload: x });
-      });
-    }
-  }, [file.url]);
-
-  const target = cache[file.url] || {
-    dependencies: [],
-    dependants: [],
-    size: 0,
-  };
-
-  const { name, version } = file.pkg;
-  const dependencies = target.dependencies.map(x => cache[x[1]]);
-  const knownDependants = target.dependants.map(x => cache[x]);
-
-  return html`
-    <${SearchInput}
-      placeholder="Search for dependencies.."
-      value=${searchTerm}
-      onChange=${setSearchTerm}
-    />
-    <div className=${styles}>
-      <div key="filesize">
-        <h2>File Size</h2>
-        <small>${target.size ? formatBytes(target.size) : 'calculating'}</small>
-      </div>
-      <${FileList}
-        title="Dependencies"
-        files=${dependencies}
-        key="dependencies"
-        packageName=${`${name}@${version}`}
-        filter=${searchTerm}
+  const file = cache['https://unpkg.com/' + request.path];
+  return (
+    !!file &&
+    html`
+      <${SearchInput}
+        placeholder="Search for dependencies.."
+        value=${searchTerm}
+        onChange=${setSearchTerm}
```

Comment:
I'm assuming we don't want this to be saved when switching between views?

Reply:
We probably should for consistency, will add it to the globalState 👍